### DISCUSSION TITLE
is there a way to 'ask' a storage if a domain object is already a part of the DB (store and commited)

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#666] [General]
Hi
I have a situation where I want to query the MS DB and ask if an object I have is already stored and commited to the MS DB.
is there a utilty for that? 
thanks

**Comment by hg-ms** [comment]
Hi,

the API you’re looking for is a bit hidden:
`embeddedStorageManager.persistenceManager().objectRegistry().lookupObjectId(Object object)` 
It will return -1 if not persisted or the assigned ObjectID if the object is persisted.

**Reply by guykatz** [reply]
@hg-ms 
similarly, assuming I take an object, that is already stored and commited, from the MS DB and I change one of its properties (simple property like int).
at this point, do I have some utilities to "ask" MS if this object is modified but not stored yet?
thanks!

**Reply by hg-ms** [reply]
There is no build in functionality to detect if an object instance has been modified after is has been persisted.
